modules/tinymce/src/plugins/accordion/test/ts/browser/AccordionPluginTest.ts

```diff
@@ -1,43 +1,63 @@
+import { Keys } from '@ephox/agar';
 import { describe, it } from '@ephox/bedrock-client';
-import { TinyHooks, TinySelections, TinyAssertions } from '@ephox/wrap-mcagar';
+import { TinyHooks, TinySelections, TinyContentActions, TinyAssertions } from '@ephox/wrap-mcagar';
 import { assert } from 'chai';
 
 import Editor from 'tinymce/core/api/Editor';
 
 import AccordionPlugin from '../../../main/ts/Plugin';
 
-describe('browser.tinymce.plugins.accordion.AccordionPluginTest', () => {
-  const hook = TinyHooks.bddSetup<Editor>({
-    plugins: 'accordion',
-    add_unload_trigger: false,
-    disable_nodechange: true,
-    indent: false,
-    entities: 'raw',
-    extended_valid_elements: 'details[class|open|data-mce-open],summary[class],div[class],p',
-    base_url: '/project/tinymce/js/tinymce'
-  }, [ AccordionPlugin ]);
-
-  interface InsertAccordionTest {
-    initialContent: string;
-    initialCursor: [ number[], number ];
-    assertContent: string;
-    assertCursor: [ number[], number ];
-  }
-  const testInsertingAccordion = (editor: Editor, test: InsertAccordionTest): void => {
-    editor.setContent(test.initialContent);
-    TinySelections.setCursor(editor, ...test.initialCursor);
-    editor.execCommand('InsertAccordion');
-    TinyAssertions.assertContent(editor, test.assertContent);
-    assert.equal(editor.selection.getNode().nodeName, 'SUMMARY');
-    TinyAssertions.assertCursor(editor, ...test.assertCursor);
+const createAccordion = (
+  { open = true, summary = 'Accordion summary...', body = '<p>Accordion body...</p>' }:
+  { open?: boolean; summary?: string; body?: string } = {}): string =>
+  `<details class="mce-accordion"${open ? ` open="open"` : ''}>` +
+  `<summary class="mce-accordion-summary">${summary}</summary>` +
+  `<div class="mce-accordion-body">${body}</div></details>`;
+
+interface InsertAccordionTest {
+  initialContent: string;
+  initialCursor: [number[], number];
+  assertContent: string;
+  assertCursor: [number[], number];
+}
+const testInsertingAccordion = (editor: Editor, test: InsertAccordionTest): void => {
+  editor.setContent(test.initialContent);
+  TinySelections.setCursor(editor, ...test.initialCursor);
+  editor.execCommand('InsertAccordion');
+  TinyAssertions.assertContent(editor, test.assertContent);
+  assert.equal(editor.selection.getNode().nodeName, 'SUMMARY');
+  TinyAssertions.assertCursor(editor, ...test.assertCursor);
+};
+
+const testEvent = (editor: Editor, eventName: string, cmd: string, callback: (event: any) => void): void => {
```

Comment:
Use a proper type here not `any` either a concrete one or T at least so that it can flow though this function.